extensions/markdown-mode/interactive.lisp

```diff
@@ -0,0 +1,124 @@
+(defpackage :lem-markdown-mode/interactive
+  (:use :cl :lem)
+  (:import-from #:alexandria :if-let :when-let :curry)
+  (:export :register-block-evaluator))
+(in-package :lem-markdown-mode/interactive)
+
+(defvar *block-evaluators* (make-hash-table :test #'equal)
+  "Dispatch table for block evaluators per language.")
+
+(defmacro register-block-evaluator (language (string callback) &body body)
+  "Convenience macro to register block evaluators, wraps setf."
+  `(setf (gethash ,language *block-evaluators*)
+         (lambda (,string ,callback)
+           ,@body)))
+
+(defmacro with-constant-position ((point) &body body)
+  "This allows you to move around the point without worry."
+  `(let ((tmp (copy-point ,point)))
```

Comment:
Did you know of `with-point(s)`?

Reply:
Doesn't seem to work the same, I will play around with it for a bit though and see if I can get it to work.

I tried to replace 
```lisp
(with-constant-position (point)
  ...)
```

with 
```lisp
(with-point ((point point))
  ...)
```

Reply:
If you don't call delete-point for copy-point, the memory leaks.